cpp/ql/test/library-tests/dataflow/DefaultTaintTracking/annotate_path_to_sink/tainted.ql

```diff
@@ -19,31 +19,27 @@ predicate isSink(Element sink) {
 predicate astTaint(Expr source, Element sink) { ASTTaintTracking::tainted(source, sink) }
 
 class SourceConfiguration extends TaintedWithPath::TaintTrackingConfiguration {
-  override predicate isSink(Element e) { any() }
+  override predicate isSink(Element e) { isSinkArgument(e) }
 }
 
-predicate irTaint(Expr source, Element sink) {
-  TaintedWithPath::taintedWithPath(source, sink, _, _)
+predicate irTaint(Element source, Element sink, string tag) {
+  exists(TaintedWithPath::PathNode sinkNode, TaintedWithPath::PathNode predNode |
+    TaintedWithPath::taintedWithPath(source, _, _, sinkNode) and
+    predNode = TaintedWithPath::Private::getAPredecessor*(sinkNode) and
```

Comment:
It looks like this might be missing an association between `source` and `predNode`.  Consider the following case with two sources that reach the same `sinkNode`:
```
source1 ---> a ---> c ---> sinkNode
                   ^
source2 ---> b ---/
```
In this case I believe `predNode` could be `a`, `b` or `c` regardless of whether `source` is `source1` or `source2`.  Should we have an extra check along the lines of:
```
source = TaintedWithPath::Private::getElementFromPathNode(TaintedWithPath::Private::getAPredecessor*(predNode))
```

Reply:
You're right! That's also why there are so many additional paths in the output. Good catch!